src/main/java/com/Elessar/app/server/LogOnHandler.java

```diff
@@ -93,7 +93,7 @@ public void handle(HttpExchange he) throws IOException {
 
                 // Get the list of unread messages sent to user
                 List<Message> messages = db.find(new Message(null, userName, null, null, false));
-                sendMessages(messages, clientURL);
+                sendMessages(messages, userName);
```

Comment:
pass the pull vs. push flag to LogOnHandler and use that to decide whether to pass userName or clientURL